pkg/securitycontext/provider.go

```diff
@@ -95,3 +96,53 @@ func makeCapabilites(capAdd []api.Capability, capDrop []api.Capability) ([]strin
 	}
 	return addCaps, dropCaps
 }
+
+func determineEffectiveSecurityContext(pod *api.Pod, container *api.Container) *api.SecurityContext {
```

Comment:
@pweil- @smarterclayton @liggitt I'm starting to think this is wrong.  I think if you have `container.SecurityContext` specified at all, it should win for that container over `pod.Spec.SecurityContext`, period.  Otherwise old clients cannot reason correctly about what a container's `SecurityContext` is.